src/mpid/ch4/netmod/ucx/ucx_impl.h

```diff
@@ -105,6 +105,14 @@ MPL_STATIC_INLINE_PREFIX bool MPIDI_UCX_is_reachable_target(int rank, MPIR_Win *
     return (winattr & MPIDI_WINATTR_NM_REACHABLE) && MPIDI_UCX_WIN_INFO(win, rank).rkey != NULL;
 }
 
+MPL_STATIC_INLINE_PREFIX bool MPIDI_UCX_is_gpu_buffer(void *buffer)
```

Comment:
This could definitely be generalized and moved to a higher layer. Maybe even compiled to just `false` for builds without GPU awareness.